comp/trace/log/trace_logger.go

```diff
@@ -86,24 +87,24 @@ func (*tracelogger) Criticalf(format string, params ...interface{}) error {
 // Flush implements Component#Flush.
 func (*tracelogger) Flush() { pkglog.Flush() }
 
-func newTraceLogger(lc fx.Lifecycle, params Params, config config.Component, telemetryCollector telemetry.TelemetryCollector) (log.Component, error) {
+func newTraceLogger(lc fx.Lifecycle, params corelogimpl.Params, config config.Component, telemetryCollector telemetry.TelemetryCollector) (log.Component, error) {
 	return NewTraceLogger(lc, params, config, telemetryCollector)
 }
 
 // NewTraceLogger creates a pkglog.Component using the provided config.LogConfig
-func NewTraceLogger(lc fx.Lifecycle, params Params, config config.LogConfig, telemetryCollector telemetry.TelemetryCollector) (log.Component, error) {
-	if params.logLevelFn == nil {
+func NewTraceLogger(lc fx.Lifecycle, params corelogimpl.Params, config config.LogConfig, telemetryCollector telemetry.TelemetryCollector) (log.Component, error) {
```

Comment:
As `NewTraceLogger` is a constructor for `log.Componnent`, I think it makes sense  this code to be located in `core/log/impl` in a new package. What do you think?

Reply:
I'll update that 👍